src/syntax/ast.rs

```diff
@@ -28,7 +31,6 @@ pub struct Pattern {
 
 #[derive(Debug, PartialEq)]
 pub enum Expression {
-    Pattern(Pattern),
```

Comment:
Here, I think we need an `Element` or `PatternElement` enum with `TextElement` and `Expression` in it.